sensors.py

```diff
@@ -85,11 +110,14 @@ def __init__(self, pi: pigpio.pi) -> None:
         self.data: bytearray = bytearray()
 
     def check_sum(self, data) -> bool:
+        # check if sum of first 30 bytes is same as last 2 bytes
         return len(data) == 32 and sum(data[:30]) == int.from_bytes(data[-2:], byteorder=self.BYTEORDER)
 
     def get_data(self, data: bytearray) -> bytearray:
-        indices = [i for i, x in enumerate(data[:-1]) if x == self.start1 and data[i+1] == self.start2]
-        for i in reversed(indices):
+        # find all occurences of self.start1 and self.start2 bytes
+        indexes = [i for i, x in enumerate(data[:-1]) if x == self.start1 and data[i+1] == self.start2]
+        # find last(newest) data frame
+        for i in reversed(indexes):
```

Comment:
`return next(data[i:i+32] for i in reversed(indexes) if self.check_sum(data[i:i+32]), bytearray())`?
Please consider

Reply:
I only added comments to explain what happens and changed variable name from indices to indexes so this should probably be separate issue.
And even then I think it's much less readable than loop version without actually adding anything.

Reply:
My point was that `for` uses iterator implicitly anyways, here we just use it explicitly. In other places you use lambdas so I guess the reader should be a Python-intermediate guy at least anyways.
But I don't insist on that ;)

Reply:
I think I only use lambda in one place, it started quite simply as lambda: deque([], 5) so quite easy to understand but I had to expand it a little. Didn't think it was worth to create method just for returning tuple.
I'm against changing it to next because it only compacts code and lowers readability(at least for me on first reading).